[gettext] Update to 0.21.1

dg0yt opened the pull request:
Marks tools as unsupported on uwp. (Discussed below. Basically reflects upstream state.)

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

dg0yt committed:
Use VERSION

dg0yt committed:
Update to 0.21.1

dg0yt committed:
Update win unicode paths patch

dg0yt committed:
Revise uwp patches

dg0yt committed:
Cleanup

dg0yt committed:
Build tools for uwp in CI

dg0yt commented:
Building tools for uwp:
~~~
../.././../src/gettext-0-ede29b59d7.clean/gettext-tools/gnulib-lib/windows-spawn.c(338): error C2065: 'HANDLE_FLAG_INHERIT': undeclared identifier
../.././../src/gettext-0-ede29b59d7.clean/gettext-tools/gnulib-lib/windows-spawn.c(402): error C2065: 'HANDLE_FLAG_INHERIT': undeclared identifier
../.././../src/gettext-0-ede29b59d7.clean/gettext-tools/gnulib-lib/windows-spawn.c(406): error C2065: 'HANDLE_FLAG_INHERIT': undeclared identifier
../.././../src/gettext-0-ede29b59d7.clean/gettext-tools/gnulib-lib/windows-spawn.c(432): error C2065: 'STARTF_USESTDHANDLES': undeclared identifier
../.././../src/gettext-0-ede29b59d7.clean/gettext-tools/gnulib-lib/windows-spawn.c(497): error C2065: 'HANDLE_FLAG_INHERIT': undeclared identifier
~~~
Did they build for port version 0.21?

Do we need tools for uwp at all, or can I simply mark them as unsupported (which seems to be the case when looking at upstream)?  We could cut parts of the patches if unsupported.

dg0yt commented:
> Did [the tools] build for port version 0.21?
> 
> Do we need tools for uwp at all, or can I simply mark them as unsupported (which seems to be the case when looking at upstream)? We could cut parts of the patches if unsupported.

@BillyONeal I need feedback on uwp here.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal commented:
> @BillyONeal I need feedback on uwp here.

I pushed a merge with master to see if https://github.com/microsoft/vcpkg/pull/26866 affects this.

And between when I typed that and now the build lab says "no" :)

Neumann-A commented:
Even if the tools would be build on uwp they wouldn't run without additional help. uwp is a cross target so no tools required.

BillyONeal commented:
> Even if the tools would be build on uwp they wouldn't run without additional help. uwp is a cross target so no tools required.

That makes sense to me. I'm building it now to try to answer the question though.

dg0yt commented:
For uwp, the tools will need more patching. The error messages look familiar with regard to the patches. My POV:
- upstream doesn't support tools for uwp (evidence: patching)
- tools for uwp don't work for the current port version in vcpkg (I guess)
- nobody needs the tools for uwp (I guess)

BillyONeal commented:
It does work in the current version:

```
Installing 2/2 gettext:x64-uwp...
Building gettext[core,tools]:x64-uwp...
-- Downloading https://ftp.gnu.org/pub/gnu/gettext/gettext-0.21.tar.gz;https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/gettext/gettext-0.21.tar.gz -> gettext-0.21.tar.gz...
-- Extracting source C:/Dev/vcpkg/downloads/gettext-0.21.tar.gz
-- Applying patch 0002-Fix-uwp-build.patch
-- Applying patch 0003-Fix-win-unicode-paths.patch
-- Applying patch 0004-Fix-uwp-tools-build.patch
-- Applying patch rel_path.patch
-- Applying patch android.patch
-- Applying patch gettext-tools_woe32dll_gettextsrc-exports.c.patch
-- Applying patch uwp_remove_localcharset.patch
-- Using source at C:/Dev/vcpkg/buildtrees/gettext/src/0.21-f35918d7b4.clean
-- Downloading https://github.com/lexxmark/winflexbison/releases/download/v2.5.25/win_flex_bison-2.5.25.zip -> win_flex_bison-2.5.25.zip...
-- Found external ninja('1.11.0').
-- Getting CMake variables for x64-uwp
-- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.
-- Using cached msys-bash-5.1.008-1-x86_64.pkg.tar.zst.
-- Using cached msys-autoconf-2.71-3-any.pkg.tar.zst.
-- Using cached msys-diffutils-3.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-binutils-2.37-5-x86_64.pkg.tar.zst.
-- Using cached msys-libtool-2.4.7-3-x86_64.pkg.tar.zst.
-- Using cached msys-file-5.41-2-x86_64.pkg.tar.zst.
-- Using cached msys-zlib-1.2.11-1-x86_64.pkg.tar.xz.
-- Using cached msys-libbz2-1.0.8-3-x86_64.pkg.tar.zst.
-- Using cached msys-coreutils-8.32-2-x86_64.pkg.tar.zst.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-libpcre-8.45-1-x86_64.pkg.tar.zst.
-- Using cached msys-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake-wrapper-11-4-any.pkg.tar.zst.
-- Using cached msys-gawk-5.1.0-2-x86_64.pkg.tar.zst.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.1-1-x86_64.pkg.tar.zst.
-- Using cached msys-libreadline-8.1.001-1-x86_64.pkg.tar.zst.
-- Using cached msys-ncurses-6.2-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake1.16-1.16.3-3-any.pkg.tar.zst.
-- Using cached msys-perl-5.32.1-2-x86_64.pkg.tar.zst.
-- Using cached msys-libcrypt-2.1-3-x86_64.pkg.tar.zst.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-3-x86_64.pkg.tar.zst.
-- Using cached msys-findutils-4.8.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-libintl-0.21-1-x86_64.pkg.tar.zst.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-11.2.0-3-x86_64.pkg.tar.zst.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using cached msys-which-2.21-4-x86_64.pkg.tar.zst.
-- Using msys root at C:/Dev/vcpkg/downloads/tools/msys2/ca907ea16f6f2ba5
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at C:/Dev/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x64-uwp-rel
-- Building x64-uwp-rel
-- Installing x64-uwp-rel
-- Getting CMake variables for x64-uwp
-- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.
-- Using cached msys-bash-5.1.008-1-x86_64.pkg.tar.zst.
-- Using cached msys-autoconf-2.71-3-any.pkg.tar.zst.
-- Using cached msys-diffutils-3.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-binutils-2.37-5-x86_64.pkg.tar.zst.
-- Using cached msys-libtool-2.4.7-3-x86_64.pkg.tar.zst.
-- Using cached msys-file-5.41-2-x86_64.pkg.tar.zst.
-- Using cached msys-zlib-1.2.11-1-x86_64.pkg.tar.xz.
-- Using cached msys-libbz2-1.0.8-3-x86_64.pkg.tar.zst.
-- Using cached msys-coreutils-8.32-2-x86_64.pkg.tar.zst.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-libpcre-8.45-1-x86_64.pkg.tar.zst.
-- Using cached msys-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake-wrapper-11-4-any.pkg.tar.zst.
-- Using cached msys-gawk-5.1.0-2-x86_64.pkg.tar.zst.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.1-1-x86_64.pkg.tar.zst.
-- Using cached msys-libreadline-8.1.001-1-x86_64.pkg.tar.zst.
-- Using cached msys-ncurses-6.2-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake1.16-1.16.3-3-any.pkg.tar.zst.
-- Using cached msys-perl-5.32.1-2-x86_64.pkg.tar.zst.
-- Using cached msys-libcrypt-2.1-3-x86_64.pkg.tar.zst.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-3-x86_64.pkg.tar.zst.
-- Using cached msys-findutils-4.8.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-libintl-0.21-1-x86_64.pkg.tar.zst.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-11.2.0-3-x86_64.pkg.tar.zst.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using cached msys-which-2.21-4-x86_64.pkg.tar.zst.
-- Using msys root at C:/Dev/vcpkg/downloads/tools/msys2/ca907ea16f6f2ba5
-- Configuring x64-uwp-dbg
-- Building x64-uwp-dbg
-- Installing x64-uwp-dbg
CMake Warning at scripts/cmake/vcpkg_copy_pdbs.cmake:44 (message):
  Could not find a matching pdb file for:

      C:/Dev/vcpkg/packages/gettext_x64-uwp/bin/intl-8.dll

Call Stack (most recent call first):
  ports/gettext/portfile.cmake:154 (vcpkg_copy_pdbs)
  scripts/ports.cmake:147 (include)


-- Performing post-build validation
Stored binary cache: "C:\Users\billy\AppData\Local\vcpkg\archives\54\546e8b0861c00ba3b5f6e7aab3c9445932d1e1259d8a9d111f0f51f4850ab6da.zip"
Elapsed time to handle gettext:x64-uwp: 20.44 min
Total install time: 25.18 min
PS C:\Dev\vcpkg>
```

BillyONeal commented:
> nobody needs the tools for uwp (I guess)

That seems probably true. Given that CI wasn't building them and it seems like they really only make sense as a host dependency.

dg0yt committed:
tools not supported on uwp

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal committed:
Add license GPL3.0-or-later.

gettext.c has:

   This program is free software: you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 3 of the License, or
   (at your option) any later version.

dg0yt commented on ports/gettext/vcpkg.json:
```diff
@@ -3,6 +3,7 @@
   "version": "0.21.1",
   "description": "GNU gettext provides libintl and a set of tools to help produce multi-lingual messages.",
   "homepage": "https://www.gnu.org/software/gettext/",
+  "license": "GPL-3.0-or-later",
```
I noticed the missing field a few minutes ago. It is like with many GNU libs: libintl is LGPL, tools, docs etc. are GPL. We probably should use `null` here and amend the installed copyright file with a comment.
https://git.savannah.gnu.org/gitweb/?p=gettext.git;a=blob;f=gettext-runtime/COPYING;h=9045d4cd3fdc2f192776f6663ff5411ded29f8a1;hb=HEAD

Neumann-A replied:
Or we need to allow license fields in features ;)

dg0yt replied:
> Or we need to allow license fields in features ;)

... however, additive features may add license restrictions infecting other features. (ffmpeg[gpl]?)

BillyONeal replied:
> I noticed the missing field a few minutes ago. It is like with many GNU libs: libintl is LGPL, tools, docs etc. are GPL. We probably should use `null` here and amend the installed copyright file with a comment. https://git.savannah.gnu.org/gitweb/?p=gettext.git;a=blob;f=gettext-runtime/COPYING;h=9045d4cd3fdc2f192776f6663ff5411ded29f8a1;hb=HEAD

I agree. I did some digging and only saw GPL 3 all over the place but if there are other licenses we should set "null" and include the right parts. Would you be willing to fix that @dg0yt?

BillyONeal replied:
I fixed :)

BillyONeal committed:
Fix license, but better.

BillyONeal committed:
Version database
